### PR TITLE
chore(Tree): Prevent string items

### PR DESCRIPTION
**Pull request by assuncaocharles**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [ ] Include a change request file using `$ yarn change`

#### Description of changes

This PR reverts the solution from #12984 and prevents strings as Tree Items

#### Focus areas to test

(optional)

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/fluentui/pull/13016)

### CODE REVIEW

File: packages/fluentui/react-northstar/src/components/Tree/Tree.tsx [hunk 1]
```diff
@@ -48,7 +48,7 @@ export interface TreeProps extends UIComponentProps, ChildrenComponentProps {
   exclusive?: boolean;
 
   /** Shorthand array of props for Tree. */
-  items?: ShorthandCollection<TreeItemProps>;
+  items?: TreeItemProps[];
```
Comment (mnajdova) [inline packages/fluentui/react-northstar/src/components/Tree/Tree.tsx:51]: We need to support children as fn, so these typings are not quite correct. Take a look on the `ShorthandValue`, it supports alto this: `(Props<P> & { children?: P['children'] | ShorthandRenderFunction<P> })`. Maybe we should add a special typings for cases where we don't want to support react node, only objects and children as fn..

Comment (assuncaocharles) [reply packages/fluentui/react-northstar/src/components/Tree/Tree.tsx]: Created `ObjectShorthandValue` and `ObjectShorthandCollection`

File: packages/fluentui/react-northstar/src/types.ts [hunk 2]
```diff
@@ -65,8 +65,15 @@ type ReactNode = React.ReactChild | React.ReactNodeArray | React.ReactPortal | b
 export type ShorthandValue<P extends Props> =
   | ReactNode
   | (Props<P> & { children?: P['children'] | ShorthandRenderFunction<P> });
```
Comment (layershifter) [inline packages/fluentui/react-northstar/src/types.ts:67]:
```suggestion
export type ShorthandValue<P extends Props> = ReactNode | ObjectShorthandValue<P>
```
??

Comment (layershifter) [reply packages/fluentui/react-northstar/src/types.ts:67]: And it looks that we need a custom proptypes for this, too

Comment (assuncaocharles) [reply packages/fluentui/react-northstar/src/types.ts]: Added

**Comment by assuncaocharles** [comment]
Is it ok now? @mnajdova and @layershifter

**Pull request merged**